### DISCUSSION TITLE
Get world (directory) name from player

### DISCUSSION THREAD

**Discussion by duranaaron** [Minestom/Minestom#1635] [Help] [answered]
How would I go over getting the name of the current world of a player?

**Comment by Timofffee** [comment, marked as answer]
To begin with, Minestom doesn't really know what the "world" is. He knows what an "instance" is. Instances don't have a name. There is only a UUID.

The name of the world that the minecraft client sees is in Dimension, which is assigned to each instance in the format "minecraft:overworld". But you can have one dimension for multiple instances.

This leads to 2 solutions to the problem:
1. Find a way in which instances will have names (this may be an extension of the InstanceContainer class or a table with values <UUID, String> (uuid, name)).
2. If you need exactly the name of the world in the format in which it is located in dimension, you can get it from there. But again, one dimension can be specified in multiple instances, so this may not be the best option.

**Reply by duranaaron** [reply]
Thanks for your reply! I knew about the Instances situation, and I was trying to somehow get the directory name. Your solutions make much more sense though, and I’ll be using one of those.
